lib/dict.c

```diff
@@ -265,12 +266,11 @@ int rc_read_dictionary (rc_handle *rh, char const *filename)
 				fclose(dictfd);
 				return -1;
 			}
-			ifilename = namestr;
+			strncpy(ifilename, namestr, sizeof(ifilename));
 			/* Append directory if necessary */
 			if (namestr[0] != '/') {
 				cp = strrchr(filename, '/');
 				if (cp != NULL) {
-					ifilename = alloca(AUTH_ID_LEN);
 					*cp = '\0';
 					snprintf(ifilename, AUTH_ID_LEN, "%s/%s", filename, namestr);
```

Comment:
we can use `sizeof(ifilename)` as you do above, to avoid copying around the size.